### PR TITLE
[tracing][feature] Support delete traces

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

Support delete traces via SDK (`pf.traces.delete`) or CLI (`pf trace delete`); valid query parameters:

- Specify `run`
- Specify `session` (non `"default"`)
- Specify `session` and `started_before` (ISO 8601)

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_cli/_pf/_trace.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+
+def add_delete_trace_params(subparsers):
+    epilog = """
+Examples:
+
+# Delete traces
+pf trace delete --run <run>
+pf trace delete --session <session>
+pf trace delete --session <session> --started-before <isoformat-string>
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_cli/_pf/_trace.py:58]: may be add a isoformat-string example here?

File: src/promptflow/promptflow/_sdk/operations/_trace_operations.py [hunk 2]
```diff
@@ -155,3 +158,48 @@ def _list_line_runs_with_runs(self, runs: typing.List[str]) -> typing.List[LineR
...
                     line_runs.append(line_run)
         return line_runs
+
+    @monitor_operation(activity_name="pf.traces.delete", activity_type=ActivityType.PUBLICAPI)
+    def delete(
+        self,
+        run: typing.Optional[str] = None,
+        session: typing.Optional[str] = None,
+        started_before: typing.Optional[str] = None,
+    ) -> int:
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/operations/_trace_operations.py:168]: give this guy some docstring

File: src/promptflow/promptflow/_sdk/operations/_trace_operations.py [hunk 3]
```diff
@@ -155,3 +158,48 @@ def _list_line_runs_with_runs(self, runs: typing.List[str]) -> typing.List[LineR
                 if line_run is not None:
                     line_runs.append(line_run)
         return line_runs
+
+    @monitor_operation(activity_name="pf.traces.delete", activity_type=ActivityType.PUBLICAPI)
+    def delete(
+        self,
+        run: typing.Optional[str] = None,
+        session: typing.Optional[str] = None,
+        started_before: typing.Optional[str] = None,
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/operations/_trace_operations.py:167]: NOT SURE IF WE SHOULD PROVIDE some systax sugar to say something like 1 day ago

Comment (zhengfeiwang) [reply src/promptflow/promptflow/_sdk/operations/_trace_operations.py:167]: thumb up as users, but thumb down as a dev... will draft some proposals on this

File: src/promptflow/promptflow/_cli/_pf/_trace.py (hunk 1 above)
Comment (zhengfeiwang) [reply src/promptflow/promptflow/_cli/_pf/_trace.py]: added

File: src/promptflow/promptflow/_sdk/operations/_trace_operations.py (hunk 2 above)
Comment (zhengfeiwang) [reply src/promptflow/promptflow/_sdk/operations/_trace_operations.py:168]: added one version

File: src/promptflow/promptflow/_sdk/operations/_trace_operations.py [hunk 4]
```diff
@@ -166,6 +166,19 @@ def delete(
         session: typing.Optional[str] = None,
         started_before: typing.Optional[str] = None,
     ) -> int:
+        """Delete traces permanently.
+
+        Support delete according to 1) run, 2) non default session and 3) session combined with started before.
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/operations/_trace_operations.py:171]: May add some example instead

**Comment by zhengfeiwang** [comment]
use #2575 to support this feature with new orm

**Pull request closed without merging**